src/Symfony/Component/HttpFoundation/InputBag.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+
+/**
+ * InputBag is a container for user input values such as $_GET, $_POST, $_REQUEST, and $_COOKIE.
+ *
+ * @author Saif Eddin Gmati <saif.gmati@symfony.com>
+ */
+final class InputBag extends ParameterBag
+{
+    /**
+     * Returns a string input value by name.
+     *
+     * @param string|null $default The default value if the input key does not exist
+     *
+     * @return string|array|null
+     */
+    public function get(string $key, $default = null)
+    {
+        if (null !== $default && !\is_string($default)) {
+            @trigger_error(sprintf('Passing a non-string value as 2nd argument to "%s::%s()" is deprecated since Symfony 5.1, pass a string or null instead.', static::class, __METHOD__), E_USER_DEPRECATED);
+        }
+
+        $value = parent::get($key, $this);
+
+        if (null !== $value && $this !== $value && !\is_string($value)) {
+            @trigger_error(sprintf('Retrieving a non-string value from "%s::%s()" is deprecated since Symfony 5.1, and will throw a "%s" exception in Symfony 6.0, use "%1$s::all()" or "%1$s::getRaw()" instead.', static::class, __METHOD__, BadRequestException::class), E_USER_DEPRECATED);
+        }
+
+        return $this === $value ? $default : $value;
+    }
+
+    /**
+     * Returns the inputs.
+     *
+     * @param string|null $key The name of the input to return or null to get them all
+     */
+    public function all(string $key = null): array
+    {
+        if (null !== $key) {
+            $value = $this->parameters[$key] ?? [];
+            if (!\is_array($value)) {
+                throw new Exception\BadRequestException(sprintf('Unexpected value for %s input.', $key));
+            }
+
+            return $value;
+        }
+
+        return $this->parameters;
+    }
+
+    /**
+     * Retrieve an input value as-is.
+     *
+     * @param mixed $default The default value if the parameter key does not exist
+     *
+     * @return mixed
+     */
+    public function getRaw(string $key, $default = null)
```

Comment:
I think this method makes no sense from an HTTP pov: the `foo=123` vs `foo[]=234` syntax is a pure PHP thing.
What about using `all()[$name]` in `HttpFoundationRequestHandler`?

Reply:
I really meant we should get rid of this method :)

Reply:
as @derrabus mentioned, there will be no method left for retriving raw value, i think doing `getRaw($key, $default)` is nicer than `all()[$key] ?? $default`

also to note, `all()[$key] ?? $default` returns `$default` when the value is null, `getRaw` does return `null` in that case.

Reply:
so its

`$request->query->getRaw($key, $value)`
<=>
`$request->query->has($key) ? $request->query->all()[$key] : $default`

Reply:
So, there is a way without.
From an API pov, this method makes no sense to me.

Reply:
there is, but not a simple or straightforward ( people might do `all()[$key] ?? $default` when they actually want `$request->query->has($key) ? $request->query->all()[$key] : $default` or the other way around ) , we don't need to make things complicated if providing a simple API is possible without breaking things. 

note: the point of `InputBag` is to make sure people don't use `get` in the wrong way, as many people are already using it as `getString`

Reply:
Ppl should know if they need `get($k)` or `all($k)` depending on the situation.
Except very generic code, nobody needs anything else.
And ppl that do need this genericity are covered.

Reply:
removed in [79a4c0f...1a3bb1a](https://github.com/symfony/symfony/compare/79a4c0f514ebcdd0587923a2aa96d8f08519f9e0..1a3bb1a83638bb00bd1d65db693ce7d91752af92), but pretty sure someone will complain about not having a method to access raw values :man_shrugging: